### PR TITLE
fix: soidity frameworks folders

### PR DESCRIPTION
**Pull request by rin-st**
Sorry @carletex I didn't notice you self-assigned the issue. 😨 

`-> Only foundry or hardhat folders should be copied / merged if selected.` - this is fixed

Regarding
`-> Ideally "none" option should not be present (or disabled). Maybe a config file in the 3rd party extension?`

Since we show prompts before reading extension, it will require to change flow.

I think we have two simpler options
1. In README of extension add explicit flag `--solidity-framework` and show both options (for me this option is better)

---
```
npx create-eth@latest -e erc-20 --solidity-framework hardhat
```
OR
```
npx create-eth@latest -e erc-20 --solidity-framework foundry
```
---

2. Add flag meaning that solidity framework is required, so options will be without "none"
```
npx create-eth@latest -e erc-20 --required-solidity-framework
```

Fixes https://github.com/scaffold-eth/create-eth/issues/89

### CODE REVIEW

**Comment by carletex** [comment]
lol, no worries @rin-st I was literally going to start working on it right now!! I've been thinking about it for a couple of days, but not actual coding :P

TLDR My take: the main thing is solved in this PR (Only foundry or hardhat folders should be copied / merged if selected), so once we test this PR, I'll merge and _maybe_ handle the `None` thing in another PR (not sure if it's that important)

> Since we show prompts before reading extension, it will require to change flow.
> I think we have two simpler options

If we want to follow this path, the _change of workflow_ might not be too difficult tho... instead of validating the extension in `parseArgumentsIntoOptions` we could copy it down already (or even parse it with a request to Github!) and check if foundry / hardhat folder exists. If so, we change the `rawOptions` variable (passed to `promptForMissingOptions`) where we can conditionally render/not render the Null option.

File: src/utils/parse-arguments-into-options.ts [hunk 1]
```diff
@@ -24,7 +24,7 @@ const validateExternalExtension = async (
       );
       await fs.promises.access(`${externalExtensionsDirectory}/${extensionName}`);
     } catch {
-      throw new Error(`Extesnion not found in "externalExtensions/${extensionName}"`);
+      throw new Error(`Extension not found in "externalExtensions/${extensionName}"`);
```
Comment (carletex) [inline src/utils/parse-arguments-into-options.ts:27]: sorry for adding this here, it's been bugging me for a while :D

File: src/tasks/copy-template-files.ts [hunk 2]
```diff
@@ -96,7 +100,14 @@ const copyExtensionsFiles = async ({ dev: isDev }: Options, extensionPath: strin
...
         const isPackageJson = isPackageJsonRegex.test(path);
-        const shouldSkip = isArgs || isTemplate || isPackageJson;
+
+        const unselectedSolidityFrameworks = [SOLIDITY_FRAMEWORKS.FOUNDRY, SOLIDITY_FRAMEWORKS.HARDHAT].filter(
+          sf => sf !== solidityFramework,
+        );
+        const isUselectedSolidityFrameworksRegexes = unselectedSolidityFrameworks.map(sf => new RegExp(`${sf}$`));
+        const isUnselectedSolidityFramework = isUselectedSolidityFrameworksRegexes.some(sfregex => sfregex.test(path));
+
+        const shouldSkip = isArgs || isTemplate || isPackageJson || isUnselectedSolidityFramework;
```
Comment (carletex) [inline src/tasks/copy-template-files.ts:110]: The empty (unselected) folder is still being created for me (with no files tho!)

image

**Review by damianmarti — APPROVED** [review]
@rin-st Thanks!!

It worked great for me. No hardhat folder is created if I select foundry and no foundry folder is created if I choose hardhat!

File: src/tasks/copy-template-files.ts (hunk 2 above)
Comment (damianmarti) [reply src/tasks/copy-template-files.ts:110]: It worked ok for me, no empty folder created.

**Comment by damianmarti** [comment]
@rin-st @carletex I think that when installing an extension that has a contract, it doesn't make sense to not select any solidity framework, so I think this is not so important.

We can add a check that if the extension you are installing has something inside hardhat or foundry, you can't select **none** as solidity framework.

File: src/tasks/copy-template-files.ts (hunk 2 above)
Comment (rin-st) [reply src/tasks/copy-template-files.ts]: [f5155f5](https://github.com/scaffold-eth/create-eth/pull/90/commits/f5155f54d7dcafb855a83c59fdd158fc602bd719) fixed, in dev mode `linkRecursive` function created folders without checking filters

**Comment by rin-st** [comment]
Regarding "none" option for solidity frameworks, I think it's better to create a new PR, what do you think?

File: src/tasks/copy-template-files.ts (hunk 2 above)
Comment (carletex) [reply src/tasks/copy-template-files.ts]: Working great now, thanks for the fix!!

**Review by carletex — APPROVED** [review]
Good stuff @rin-st !

Yeah, let's do the "none" in another PR. Feel free to tackle it directly... or create an issue for it.

**Comment by carletex** [comment]
Created https://github.com/scaffold-eth/create-eth/issues/93
